src/dp_abstraction/aca_dataplane_ovs.cpp

```diff
@@ -72,6 +74,35 @@ static bool aca_lookup_subnet_info(GoalState &parsed_struct, const string target
   return false;
 }
 
+static bool aca_lookup_auxgateway_info(GoalState &parsed_struct, const string targeted_vpc_id,
+                                       AuxGateway &found_auxgateway)
+{
+  // TODO: cache the subnet information to a dictionary to provide
+  // a faster look up for the next run, only use the below loop for
+  // cache miss.
+  // Look up the subnet configuration to query for tunnel_id
+  for (int j = 0; j < parsed_struct.vpc_states_size(); j++) {
```

Comment:
let's start with int i=0 instead of j, I know you get it from aca_lookup_subnet_info, please go fix there also :)